app/receiver/index.ts

```diff
@@ -0,0 +1,177 @@
+  )
+  const getObjectParams = {
+    Bucket: bucket,
+    Key: key,
+  }
+  // Get data from S3
+  const object = await s3.send(new GetObjectCommand(getObjectParams))
+  if (!object.Body) throw new Error('S3 object has no body')
+
+  const parsed = await simpleParser(object.Body as unknown as Source)
+
+  const emailModel = {
+    from: parsed.from,
+    text: parsed.text,
+    subject: parsed.subject,
+  }
+
+  if (!emailModel.from) throw new Error('S3 object has no sender')
+
+  const userEmail = emailModel.from.value[0].address
+  if (!userEmail)
+    throw new Error(
+      'Error parsing sender email from model: ' +
+        JSON.stringify(emailModel.from)
+    )
+
+  if (!emailModel.subject || !emailModel.text) {
+    await sendEmail(
+      userEmail,
+      'Invalid GCN Circular Submission',
+      'Subject and Body are required to submit a GCN Circular via email.'
+    )
+    return
+  }
+
+  // Check if submitter is in the submitters group
+  var listUsersParams = {
+    GroupName: 'gcn.nasa.gov/circular-submitter',
+    UserPoolId: USER_POOL_ID,
+  }
+
+  const data = await client.send(new ListUsersInGroupCommand(listUsersParams))
+
+  const userData = data.Users?.filter(
+    (user) =>
+      user.Attributes?.find((attr) => attr.Name == 'email')?.Value == userEmail
+  ).map((user) => ({
+    sub: extractAttributeRequired(user, 'sub'),
+    email: extractAttributeRequired(user, 'email'),
+    name: extractAttribute(user, 'name'),
+    affiliation: extractAttribute(user, 'custom:affiliation'),
+  }))
+
+  if (!userData || userData.length == 0) {
+    await sendEmail(
+      userEmail,
+      'GCN Circular Submission Warning: Missing permissions',
+      'You do not have the required permissions to submit GCN Circulars. If you believe this to be a mistake, please fill out the form at https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn, and we will look into resolving it as soon as possible.'
+    )
+    return
+  }
+
+  // Validate Subject Line
+  if (emailIsAutoReply(emailModel.subject)) {
+    return
+  }
+
+  if (!subjectIsValid(emailModel.subject)) {
+    await sendEmail(
+      userEmail,
+      'GCN Circular Submission Warning: Invalid subject structure',
+      'The submission of your Circular has been rejected, as the subject line does not conform to the appropriate format. Please see https://gcn.nasa.gov/circulars/classic#submission-process for more information.'
+    )
+    return
+  }
+
+  const circular = {
+    dummy: 0,
+    createdOn: Date.now(),
+    subject: emailModel.subject,
+    body: emailModel.text,
+    sub: userData[0].sub,
+    submitter: formatAuthor(userData[0]),
+  }
+
+  // Get tables for autoincrement
+  const db = await tables()
+  const tableName = db.name('circulars')
+  const counterTableName = db.name('auto_increment_metadata')
+
+  if (!tableName || !counterTableName) {
+    throw new Error('Could not find tables')
+  }
+
+  const doc = DynamoDBDocument.from(dynamodbClient)
```

Comment:
Move this up to module scope. Also, don't define dynamodbClient because we never use it directly.